toolchain/docs/lower.md

```diff
@@ -23,3 +24,26 @@ can first generate type information for function arguments.
 Lowering is done per `SemIR::InstBlock`. This minimizes changes to the
 `IRBuilder` insertion point, something that is both expensive and potentially
 fragile.
+
+## Mangling
+
+Part of lowering is choosing deterministically unique identifiers for each
+lowered entity to use in platform object files. Any feature of an entity (eg:
+the namespace it appears in, parameters for overloaded functions, etc) that
+would create a distinct entity must be included in some way in the generated
+identifier.
+
+The current rudimentary name mangling scheme is as follows:
+
+-   Start with `_C`
```

Comment:
Had you considered a sequence-based ordering versus "start with" and "then"? e.g.:

```
1.`_C` prefixes all names.
2. The unqualified function name.
3. Scopes.
```

Reply:
Yeah, I've given that a go.